### PR TITLE
add certificate template support for google-cas installation

### PR DESCRIPTION
**Pull request by liwenhao0810**
Add certificate template support by appending the CT to the --ca_pool param.

### CODE REVIEW

**Comment by zerobfd** [comment]
The code directionally looks good, but we don't edit the `asmcli` file directly. In the /asmcli directory there are several files that get merged to create the final tool. See https://github.com/GoogleCloudPlatform/anthos-service-mesh-packages/blob/main/docs/developer-guide.md for more details on how to work with the repo.

File: asmcli/asmcli
Comment (shankgan) [inline asmcli/asmcli:1244]: nit: "if certificate template needs to be specified"

File: asmcli/asmcli
Comment (shankgan) [inline asmcli/asmcli:1624]: @pmansour - do you for-see any issues with using ":" as a delimiter to specify both the ca-pool and the certificate template in the same variable?

File: asmcli/asmcli
Comment (shankgan) [inline asmcli/asmcli:1605]: nit: maybe add an additional check to see if there is more than one ":' in the string, this errors out as an unacceptable format.

File: asmcli/asmcli
Comment (lei-tang) [inline asmcli/asmcli:2091]: nit: to be consistent with the comment above "# check if CA_NAME is ca_pool:cert_template format", consider adding a comment here.

File: asmcli/asmcli
Comment (liwenhao0810) [reply asmcli/asmcli:1605]: Added in validate_private_ca().

File: asmcli/components/ca/private-ca.sh
Comment (zerobfd) [inline asmcli/components/ca/private-ca.sh:10]: Failing lint:
```
In asmcli line 2084:
  elif [[ 0 < $(grep -o ":" <<< "${CA_NAME}" | wc -l) ]]; then
            ^-- SC2071: < is for string comparisons. Use -lt instead.
```

Comment (liwenhao0810) [reply asmcli/components/ca/private-ca.sh]: fixed.

File: asmcli/components/ca/private-ca.sh
Comment (pmansour) [inline asmcli/components/ca/private-ca.sh:17]: suggestion: you'd get much simpler bash error checks here and more accurate validations if you make the above template strings regexes, rather than trying to count the number of slashes and colons in each occurrence. For example, despite all the complex / hard-to-read validation logic here, it seems that the string "projects/foo/coffeeShops/CafeVita/espressoMachines/LaMarzocco" would be considered a valid CaPool.

File: asmcli/commands/install.sh
Comment (liwenhao0810) [reply asmcli/commands/install.sh]: fixed

File: asmcli/commands/install.sh
Comment (liwenhao0810) [reply asmcli/commands/install.sh]: fixed

File: asmcli/components/ca/private-ca.sh
Comment (liwenhao0810) [reply asmcli/components/ca/private-ca.sh]: fixed using regex check.

File: asmcli/commands/install.sh
Comment (liwenhao0810) [reply asmcli/commands/install.sh:93]: Not sure how to check whether have the privileges to bind IAM roles, one thought here is that even it fails to bind the roles, it will not fail the installation. So from that perspective this is something user could work around.
But agreed on that is right things to do as following, if you know the command that check whether user have the permission to set the bindings, please let me know.

**Review by zerobfd — APPROVED** [review]
LGTM when the current threads are resolved

File: asmcli/commands/help.sh
Comment (shankgan) [reply asmcli/commands/help.sh:117]: @zerobfd - the UX isn't ideal as implemented now; we are combining two different parameters into the same flag. Would it be acceptable to add an extra flag to asmcli (--ca-template)?

File: asmcli/commands/install.sh
Comment (shankgan) [reply asmcli/commands/install.sh:99]: A good compromise IMO would be to gate ALL the IAM role changes with the "can_modify_gcp_iam_roles" function. I think adding checks here could get complicated. How about we make the required IAM roles more easily available in our documentation?

File: asmcli/commands/help.sh
Comment (zerobfd) [reply asmcli/commands/help.sh:117]: As long as the UX is decided before we merge anything into staging. Also please keep in mind that `asmcli` is going away as soon as humanly possible as soon as the API installation reaches parity, so any new args/changes is going to introduce more work for CAS customers to use and then migrate off of.

Comment (liwenhao0810) [reply asmcli/commands/help.sh:117]: Agreed with keep as this for now and see if we need to add a flag here or directly in API installation.

File: asmcli/commands/install.sh
Comment (liwenhao0810) [reply asmcli/commands/install.sh]: Added the conditional check "can_modify_gcp_iam_roles".

**Pull request merged**
